operator/cmd/mesh/manifest-apply.go

```diff
@@ -213,3 +225,32 @@ func ApplyManifests(setOverlay []string, inFilenames []string, force bool, dryRu
 
 	return nil
 }
+
+// DetectIstioVersionDiff will show warning if istioctl version and control plane version are different
+// nolint: interfacer
+func DetectIstioVersionDiff(cmd *cobra.Command, obv string, maArgs *manifestApplyArgs) error {
+	var opts clioptions.ControlPlaneOptions
+	icps, err := GetVersionInfo(maArgs.kubeConfigPath, maArgs.context, opts)
+	if err != nil {
+		return err
+	}
+	if len(*icps) != 0 {
+		var icpVer string
+		for _, icp := range *icps {
+			icpVer = icp.Info.Version
+		}
+		if obv != icpVer {
+			cmd.Printf("! istioctl version %s and the control plane version %s are different.\n", obv, icpVer)
+		}
+	}
+	return nil
+}
+
+// GetVersionInfo returns control plan version
+func GetVersionInfo(kubeConfigPath string, context string, opts clioptions.ControlPlaneOptions) (*istioVersion.MeshInfo, error) {
+	kubeClient, err := kubernetes.NewExtendedClient(kubeConfigPath, context, opts)
+	if err != nil {
+		return nil, err
+	}
+	return kubeClient.GetIstioVersions(controller.IstioNamespace)
```

Comment:
This command is super slow on anything other than the smallest clusters. Like over 30s

Reply:
I see, but I'm not sure if there is another better alternative available. `GetIstioVersions` is currently used for.
1. istioctl [version cmd](https://github.com/istio/istio/blob/master/istioctl/cmd/version.go#L70)
2. istioctl [upgrade cmd](https://github.com/istio/istio/blob/master/operator/cmd/mesh/upgrade.go#L273). 

> We must use the Control Plane Version reported by istioctl version in the message.

It's also recommended by @esnible in [the comment](https://github.com/istio/istio/issues/18487#issuecomment-593441401)

Reply:
I believe there is a bit of performance improvement due to a recent update https://github.com/istio/istio/pull/24903 in getting the control plan version using `:15014/version` on pilot ?